### PR TITLE
AIMS-334 Add function to delete transfers, and add date/time to active transfer list

### PR DESCRIPTION
**Pull request by rfox2**
What: Added service class for destroying transfers, added controller method for doing so, and added a date / time stamp to the active transfer list

### CODE REVIEW

File: app/controllers/transfers_controller.rb [hunk 1]
```diff
@@ -41,6 +41,15 @@ def create
     redirect_to transfer_path(id: transfer.id)
   end
 
+  def destroy
+    if DestroyTransfer.call(Transfer.find(params[:id]), current_user)
+      flash[:notice] = "Transfer canceled."
+    else
+      flash[:error] = "Transer not canceled due to system error."
```
Comment (jonhartzler) [inline app/controllers/transfers_controller.rb]: Perhaps: 

System Error.  The transaction has not been canceled.   Please try again and if it is unsuccessful --insert text about what they can do.

File: app/services/destroy_request.rb [hunk 2]
```diff
@@ -15,4 +15,12 @@ def destroy
     ActivityLogger.remove_request(request: request, user: user)
     status
   end
+
+  def self.shelve_trays(shelf, user)
```
Comment (jonhartzler) [inline app/services/destroy_request.rb]: Should this method have spec?

File: app/controllers/transfers_controller.rb (hunk 1 above)
Comment (rfox2) [inline app/controllers/transfers_controller.rb]: I'll rewrite this message

File: app/services/destroy_request.rb (hunk 2 above)
Comment (rfox2) [inline app/services/destroy_request.rb]: Yes, probably. I'll add it.

File: app/services/destroy_transfer.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  def initialize(transfer, user)
+    @transfer = transfer
+    @shelf = transfer.shelf
+    @user = user
+  end
+
+  def destroy
+    status = @transfer.destroy!
```
Comment (jgondron) [inline app/services/destroy_transfer.rb]: Can we wrap this in a transaction block? If anything happens with reshelving any of the trays, or with destroying the transfer, we would end up in a partial state.

Comment (rfox2) [inline app/services/destroy_transfer.rb]: Done

**Comment by jgondron** [comment]
:+1:

**Comment by jonhartzler** [comment]
:+1:

**Pull request merged**